dispatch.go

```diff
@@ -89,22 +90,35 @@ func newDispatch(enc encoder, dec byteReadingDecoder, l LogInterface, wef WrapEr
 
 type call struct {
 	ch             chan error
+	doneCh         chan struct{}
 	method         string
 	seqid          int
 	arg            interface{}
 	res            interface{}
 	errorUnwrapper ErrorUnwrapper
 	profiler       Profiler
+	ctx            context.Context
```

Comment:
Would you go that far, though? Looking at all the args in call, it seems like it's much more than a context!